Adding fields

gpongelli opened the pull request:
Explicitly adding all the MASK_ENABLE bits, MANUFACTURER and REVISION reg.

gpongelli committed:
Adding fields

Explicitly adding all the MASK_ENABLE bits, MANUFACTURER and REVISION registry

siddacious commented on adafruit_ina260.py:
```diff
@@ -195,27 +207,31 @@ def __init__(self, i2c_bus, address=0x40):
     mask_enable = RWBits(16, _REG_MASK_ENABLE, 0, 2, False)
     alert_limit = RWBits(16, _REG_ALERT_LIMIT, 0, 2, False)
+    
+    manufacturer_id = ROUnaryStruct(_REG_MFG_UID, ">H")
```
I don't think these three are particularly useful for the user. If you'd like to keep them, please add them as a check in `__init__` that verifies we're talking to the correct chip. 

You can see an example here:
 https://github.com/adafruit/Adafruit_CircuitPython_LSM6DS/blob/master/adafruit_lsm6ds/__init__.py#L206
 
With that in mind, I don't thing we need the manufacturer id or die id. Please remove them and rename `device_id` to `_device_id` to make it clear it's not meant for public consumption.

gpongelli replied:
interesting point, I'm going to add this check

siddacious reviewed (changes requested):
This is failing because there is not documentation for the new code. This will need to be fixed to merge.

Secondly, if you are going to add new features, please add one or more examples that show how to use the new features.

siddacious commented:
Thank you for your contribution! This particular library is near and dear to me, so it's great to see someone else chipping in!

gpongelli committed:
Adding check on manufacturer and device id

gpongelli committed:
adding system reset bit

gpongelli committed:
documentation

gpongelli commented:
Thank you for this library! 

I'm working on this chip with a custom board (not the Adafruit one) and this library is helping me a lot, instead of writing similar code to yours 😃

gpongelli committed:
removing space only rows

gpongelli committed:
fixing some lint check

I cannot fix errors on CRLF if any cause I'm working on windows.

siddacious commented:
It looks like you're to the Black formatting check, almost there!
See here for more info:
https://learn.adafruit.com/improve-your-code-with-pylint/black

gpongelli committed:
black executed

gpongelli committed:
add missing import

gpongelli commented:
@siddacious sorry but I do not understand what's still missing, could you help me ?

gpongelli committed:
adding static method to retrieve values

giving register's content e.g.  ConversionTime.get_seconds(ConversionTime.TIME_332_us) will return 0.000332 .

similarly, ConversionTime.get_seconds(obj.voltage_conversion_time) will return the value reading from registry

siddacious commented:
@gpongelli Please try to reproduce the sphinx build on your machine to see if you get the same errors.

You can use his guide for reference:
https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/testing-with-github-actions

gpongelli committed:
fix sphinx-build

gpongelli committed:
fix for pylint

gpongelli committed:
remove trailing space

gpongelli commented:
@siddacious was the pipeline executed before my branch did start ?
because I'm doing fixes over fixes on file part I've not touched, like the following rows from [master commit](https://github.com/adafruit/Adafruit_CircuitPython_INA260/blob/master/adafruit_ina260.py#L43)
```
**Hardware:**

* `INA260 Breakout <https://www.adafruit.com/products/4226>`_

**Software and Dependencies:**

* Adafruit CircuitPython firmware for the supported boards:
  https://github.com/adafruit/circuitpython/releases

 * Adafruit's Bus Device library: https://github.com/adafruit/Adafruit_CircuitPython_BusDevice

 * Adafruit's Register library: https://github.com/adafruit/Adafruit_CircuitPython_Register
```

siddacious commented:
@gpongelli You're likely running into checks that were added after the last time a build was done on this repo.

Your help updating the code would be **_much_** appreciated

gpongelli commented:
@siddacious checks pass now

siddacious commented:
@gpongelli please see my earlier request for change where I ask for an example that tests the new code. It will help me test as well as help users understand how to use the new features

gpongelli committed:
added example file

gpongelli committed:
reformat after black

gpongelli committed:
LF fixes

gpongelli commented:
@siddacious look at the new example file and let me know

gpongelli committed:
Update adafruit_ina260.py

add empty rows to resolve merge conflicts

gpongelli commented:
@siddacious @ladyada anyone there?

ladyada commented:
hihi please adjust the PR so CI passes
https://github.com/adafruit/Adafruit_CircuitPython_INA260/runs/1862521113
we have a guide on linting
https://learn.adafruit.com/improve-your-code-with-pylint
@dherrada can help if you're not sure what to do

gpongelli commented:
Yes thanks, ‘cause it’s not clear how a commit with only empty rows to fix a merge conflict can not pass the pipeline.

Has something changed into pipeline’s checks during last month?


> Il giorno 15 feb 2021, alle ore 18:38, ladyada <notifications@github.com> ha scritto:
> 
> ﻿
> hihi please adjust the PR so CI passes
> https://github.com/adafruit/Adafruit_CircuitPython_INA260/runs/1862521113
> we have a guide on linting
> https://learn.adafruit.com/improve-your-code-with-pylint
> @dherrada can help if you're not sure what to do
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

ladyada commented:
yes we now require copyright deets!

gpongelli committed:
added copyright

gpongelli commented:
@ladyada copyrights added after rebase, but still fails due to something runner-related issue.

ladyada commented:
weird - dunno what that error is. 
@tannewt @dherrada plz take a look when yr around next

evaherrada commented:
@ladyada @gpongelli Looks like an issue with the latest Pylint (https://github.com/PyCQA/pylint/issues/4097) 2.6.1. Should be fixed by just running the CI again ~~when 2.6.2 gets released~~

Edit: looks like 2.6.2 has been released.

evaherrada commented:
Yeah, looks like it was just a bug in 2.6.1. Seems to be fixed now.

gpongelli commented:
Pipeline now passes. 

There’s still an old code review request already addressed.

ladyada commented:
aawesome

gpongelli commented:
thanks for merging this PR.

is it planned a 1.2.7 version sooner or later ?

ladyada commented:
there are version release sweeps once a week

evaherrada commented:
@gpongelli Yep. This should be released tomorrow.
